# Investment Strategy for Telcov

## *Problem 3*: Country Analysis
***

### *Summary based on the data analysis*
***

<b> The top 9 countries by highest total funding are `['USA','CHN','GBR','IND','CAN','FRA','ISR','DEU','JPN']`
 <span style='color:Blue'> USA, Chine, Great Britain, India, Canada, France, Israel, Germany and Japan. </span>
</b>

<span style='color:Green'> <b>
    -  Top English Speaking Country: USA (United States of America)
    -  Second English Speaking Country: GBR (Great Britain) and  
    -  Third English Speaking Country: IND (India)  
    
 ## *Problem 4*: Sector Analysis 1
 ***
    
<span style='color:Green'> <b>
    There were many sub sectors in mapping file which were misspelled. This resulted in a lot of unmapped main sectors (over 3000)for the sub sectors when the merge was done. After correcting all the spelling mistakes, there were only 15 sub sectors which were not mapped. As the data was was still very close to the original and all main sectors were correctly mapped, the exercise to map the remaining 15 sub sectors was dropped. I also removed the main sector Blanks as we want to work with only the 8 main sectors including others.</b></span>
    
    The cleaned data is saved as "mainsectors.csv"
***   
 

## Code

In [2]:
#system specific
import os
import sys
assert sys.version_info >= (3,5)

#import the needed libraries
import numpy as np
import pandas as pd

#visualizations
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

#handle unwanted warnings
import warnings
warnings.filterwarnings(action='ignore',category=DeprecationWarning)
warnings.filterwarnings(action='ignore',category=FutureWarning)

#consistent size plot
from pylab import rcParams
rcParams['figure.figsize']= 12,5
rcParams['xtick.labelsize']= 12
rcParams['ytick.labelsize']= 12
rcParams['axes.labelsize']= 12

#enable viewing of all the columns in a df
pd.options.display.max_columns = None

In [3]:
#load the chosen investment type data
df =  pd.read_csv('ventures.csv',delimiter=',',encoding='ISO-8859-2')
df.head()

permalink                    name          homepage_url  \
0                   fame                   #fame    http://livfame.com   
1   the-one-of-them-inc-  (THE) ONE of THEM,Inc.   http://oneofthem.jp   
2                0-6-com                 0-6.com    http://www.0-6.com   
3  0ndine-biomedical-inc  Ondine Biomedical Inc.  http://ondinebio.com   
4                 0xdata                  H2O.ai        http://h2o.ai/   

       category_list     status country_code state_code       region  \
0              Media  operating          IND         16       Mumbai   
1  Apps|Games|Mobile  operating          NaN        NaN          NaN   
2        Curated Web  operating          CHN         22      Beijing   
3      Biotechnology  operating          CAN         BC    Vancouver   
4          Analytics  operating          USA         CA  SF Bay Area   

            city  founded_at      company_permalink  \
0         Mumbai         NaN                   fame   
1            NaN         NaN   the-one-of-them-inc-   
2        Beijing  01-01-2007                0-6-com   
3      Vancouver  01-01-1997  0ndine-biomedical-inc   
4  Mountain View  01-01-2011                 0xdata   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/650b8f704416801069bb178a1418776b            venture   
2  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
3  /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
4  /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015         10000000.0  
1                  B  30-01-2014          3406878.0  
2                  A  19-03-2008          2000000.0  
3                NaN  21-12-2009           719491.0  
4                  B  09-11-2015         20000000.0

In [4]:
df['country_code'].unique()

array(['IND', nan, 'CHN', 'CAN', 'USA', 'GBR', 'FRA', 'ROM', 'KOR', 'IRL',
       'SWE', 'SGP', 'NLD', 'RUS', 'BEL', 'ESP', 'HUN', 'JPN', 'DEU',
       'ITA', 'HKG', 'BRA', 'FIN', 'CHE', 'PRT', 'SVN', 'THA', 'DNK',
       'TWN', 'ISR', 'NOR', 'LTU', 'ISL', 'MEX', 'AUS', 'NZL', 'AUT',
       'ARG', 'PHL', 'MNE', 'MYS', 'TUR', 'POL', 'GGY', 'EST', 'LBN',
       'GRC', 'IDN', 'CYP', 'SVK', 'ARE', 'EGY', 'ARM', 'ZAF', 'TUN',
       'KEN', 'COL', 'CZE', 'CMR', 'PRI', 'CYM', 'PER', 'ECU', 'CHL',
       'VNM', 'URY', 'LVA', 'PAK', 'NGA', 'HRV', 'LUX', 'UKR', 'BMU',
       'MUS', 'BGR', 'TTO', 'PAN', 'MMR', 'JOR', 'KAZ', 'MAR', 'LIE',
       'GTM', 'SAU', 'TAN', 'KNA', 'SEN', 'MLT', 'MCO', 'BAH', 'GHA',
       'KWT', 'UGA', 'LAO', 'BGD', 'BWA', 'MAF', 'GIB'], dtype=object)

In [5]:
#check the number of NaN in the country code
len(df[df['country_code'].isna()])

1692

In [6]:
df[df['country_code'].isna()].head(10)

permalink                    name  \
1    the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
28              1234enter               1234ENTER   
31           123greetings            123Greetings   
46               17-media                17 Media   
56            1800diapers             1800Diapers   
150               360quan                 360Quan   
151               360quan                 360Quan   
152               360quan                 360Quan   
189               3dvista                 3DVista   
232         4-antibody-ag           4-Antibody AG   

                     homepage_url  \
1             http://oneofthem.jp   
28    http://www.1234enter.com.br   
31   http://www.123greetings.com/   
46                http://17.media   
56                            NaN   
150        http://www.360quan.com   
151        http://www.360quan.com   
152        http://www.360quan.com   
189        http://www.3dvista.com   
232                           NaN   

                                         category_list     status  \
1                                    Apps|Games|Mobile  operating   
28   Advertising|Analytics|Business Services|E-Comm...  operating   
31                                            Internet  operating   
46                                                 NaN  operating   
56                               Baby Accessories|Kids  operating   
150         Internet|Social Media|Social Network Media  operating   
151         Internet|Social Media|Social Network Media  operating   
152         Internet|Social Media|Social Network Media  operating   
189                                        Curated Web  operating   
232                                                NaN  operating   

    country_code state_code region city  founded_at     company_permalink  \
1            NaN        NaN    NaN  NaN         NaN  the-one-of-them-inc-   
28           NaN        NaN    NaN  NaN  01-01-2012             1234enter   
31           NaN        NaN    NaN  NaN  01-05-1997          123greetings   
46           NaN        NaN    NaN  NaN  15-07-2015              17-media   
56           NaN        NaN    NaN  NaN         NaN           1800diapers   
150          NaN        NaN    NaN  NaN  28-03-2006               360quan   
151          NaN        NaN    NaN  NaN  28-03-2006               360quan   
152          NaN        NaN    NaN  NaN  28-03-2006               360quan   
189          NaN        NaN    NaN  NaN         NaN               3dvista   
232          NaN        NaN    NaN  NaN         NaN         4-antibody-ag   

                             funding_round_permalink funding_round_type  \
1    /funding-round/650b8f704416801069bb178a1418776b            venture   
28   /funding-round/6343322ed785252d4c5ada444fa7d8e4            venture   
31   /funding-round/5dd1821a92088ff96b467eb16431731a            venture   
46   /funding-round/f8ffde9fa822843ac6ea9b66cc615df1            venture   
56   /funding-round/5e2b8cd8ad0dcdff961fd09dee78d487            venture   
150  /funding-round/67f5b12f4676395df03f32378bcfebae            venture   
151  /funding-round/97d556581f9645328053142de8fa394d            venture   
152  /funding-round/e44a57bae2788e5ac86c81f9e6d2b256            venture   
189  /funding-round/689a5e7f3cea9622f820996a11248b78            venture   
232  /funding-round/7c1b312fd45a3ac21a44a9c8b2f6fb0f            venture   

    funding_round_code   funded_at  raised_amount_usd  
1                    B  30-01-2014          3406878.0  
28                 NaN  22-11-2013           150267.0  
31                 NaN  23-07-2015          5400000.0  
46                   A  05-11-2015         10000000.0  
56                   B  04-12-2007          7000000.0  
150                  B  05-12-2007         10000000.0  
151                  C  01-07-2008          8000000.0  
152                  A  08-02-2006          5000000.0  
189                NaN  03-06-2008           475000.0  
232                  A  16-01-

Looking at the few rows, it is highly unlikely to get a good estimate of the country code. Imputing the country code would be very risky as no indication is available from the other columns in the dataset. Sadly, this 1692 rows will have to be dropped again. Moreover, Telcov is very clear that it wants to invest only in the countries where English is the official language. 

In [7]:
null_country_index = df[df['country_code'].isna()].index

In [8]:
#drop these rows
df.drop(null_country_index,axis=0,inplace=True)

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 44949 entries, 0 to 46640
Data columns (total 16 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   permalink                44949 non-null  object 
 1   name                     44949 non-null  object 
 2   homepage_url             42966 non-null  object 
 3   category_list            44748 non-null  object 
 4   status                   44949 non-null  object 
 5   country_code             44949 non-null  object 
 6   state_code               44253 non-null  object 
 7   region                   44400 non-null  object 
 8   city                     44400 non-null  object 
 9   founded_at               37443 non-null  object 
 10  company_permalink        44949 non-null  object 
 11  funding_round_permalink  44949 non-null  object 
 12  funding_round_type       44949 non-null  object 
 13  funding_round_code       25075 non-null  object 
 14  funded_at             

In [10]:
#list the top countries based on their frequency in the chosen investment type
df['country_code'].value_counts().sort_values(ascending=False)[:10]

USA    33745
GBR     1904
CHN     1523
CAN     1153
FRA      864
IND      797
ISR      746
DEU      528
ESP      291
JPN      266
Name: country_code, dtype: int64

Amongst these, USA, GBR, CAN, IND have English as the official language. However, the goal is to check the top 9 countries in terms of the highest total funding.

In [11]:
#list the top 9 countries by highest total funding
df.groupby('country_code')['raised_amount_usd'].sum().sort_values(ascending=False)[:9]

country_code
USA    3.999351e+11
CHN    3.910177e+10
GBR    1.824134e+10
IND    1.405060e+10
CAN    8.825487e+09
FRA    6.775044e+09
ISR    6.503021e+09
DEU    6.061141e+09
JPN    3.242032e+09
Name: raised_amount_usd, dtype: float64

In [12]:
#create a dataframe called top9
top9_country_code = ['USA','CHN','GBR','IND','CAN','FRA','ISR','DEU','JPN']
top9 = df[df['country_code'].isin(top9_country_code)]

In [13]:
top9.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 41526 entries, 0 to 46640
Data columns (total 16 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   permalink                41526 non-null  object 
 1   name                     41526 non-null  object 
 2   homepage_url             39635 non-null  object 
 3   category_list            41349 non-null  object 
 4   status                   41526 non-null  object 
 5   country_code             41526 non-null  object 
 6   state_code               41257 non-null  object 
 7   region                   41107 non-null  object 
 8   city                     41107 non-null  object 
 9   founded_at               34687 non-null  object 
 10  company_permalink        41526 non-null  object 
 11  funding_round_permalink  41526 non-null  object 
 12  funding_round_type       41526 non-null  object 
 13  funding_round_code       23222 non-null  object 
 14  funded_at             

The top9 countries by total investment type constitutes <span style='color:Blue'><b>92%</b></span> of the total dataset in initial dataframe that was read on the ventures investment type !

In [14]:
top9.groupby('country_code')['raised_amount_usd'].sum().sort_values(ascending=False)

country_code
USA    3.999351e+11
CHN    3.910177e+10
GBR    1.824134e+10
IND    1.405060e+10
CAN    8.825487e+09
FRA    6.775044e+09
ISR    6.503021e+09
DEU    6.061141e+09
JPN    3.242032e+09
Name: raised_amount_usd, dtype: float64

<span style='color:Blue'><b> USA (United States of America), GBR (Great Britain) and IND (India) and the top 3 English speaking countries which has received the highest total funding </b></span>

## *Problem 4:* Sector Analysis 1
***


In [15]:
#load the mapping file
mapping = pd.read_csv('mapping.csv',delimiter=',',encoding='ISO-8859-2')


In [16]:
mapping

category_list  Automotive & Sports  Blanks  \
0                 NaN                    0       1   
1                  3D                    0       0   
2         3D Printing                    0       0   
3       3D Technology                    0       0   
4          Accounting                    0       0   
..                ...                  ...     ...   
683         Wholesale                    0       0   
684  Wine And Spirits                    0       0   
685          Wireless                    0       0   
686             Women                    0       0   
687      Young Adults                    0       0   

     Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
0                             0              0       0              0   
1                             0              0       0              1   
2                             0              0       0              1   
3                             0              0       0              1   
4                             0              0       0              0   
..                          ...            ...     ...            ...   
683                           0              0       0              0   
684                           0              1       0              0   
685                           0              1       0              0   
686                           0              0       0              0   
687                           0              0       0              0   

     News, Search and Messaging  Others  \
0                             0       0   
1                             0       0   
2                             0       0   
3                             0       0   
4                             0       0   
..                          ...     ...   
683                           0       1   
684                           0       0   
685                           0       0   
686                           0       1   
687                           0       1   

     Social, Finance, Analytics, Advertising  
0                                          0  
1                                          0  
2                                          0  
3                                          0  
4                                          1  
..                                       ...  
683                                        0  
684                                        0  
685                                        0  
686                                        0  
687                                        0  

[688 rows x 10 columns]

In [17]:
#number of category list in the top9 dataframe
len(top9['category_list'].unique())

8350

All these need to be mapped to one of the 8 main sectors as in the mapping file. 

In [18]:
#number of unique category list in mapping which are mapped to 8 main sectors
len(mapping['category_list'].unique())

688

In [19]:
top9['category_list'][:20]

0                                Media
2                          Curated Web
3                        Biotechnology
4                            Analytics
5                            Analytics
6                            Analytics
7     Apps|Cable|Distribution|Software
8                            Education
9                          Curated Web
10                         Curated Web
11                         Curated Web
12                         Curated Web
13                         Curated Web
14        Local Businesses|Restaurants
15        Local Businesses|Restaurants
16        Local Businesses|Restaurants
17              Cosmetics|Marketplaces
18                         Games|Media
19                         Hospitality
20                           Education
Name: category_list, dtype: object

There are multiple sub-sectors in many of the rows. Business rule to be applied after aligning with the CEO of Telcov <span style='color:Red'> <b> that the first string
before the vertical bar will be considered the primary sector </b>


In [20]:
#check NaNs in the category list and replace with the text Blanks
len(top9[top9['category_list'].isna()])

177

In [21]:
top9[top9['category_list'].isna()].head()

permalink               name                    homepage_url  \
30        123feng-com        123Feng.Com             http://123feng.com/   
63            1mxians            1mxians          http://www.1mxian.com/   
207       3pl-central        3PL Central          http://3plcentral.com/   
285             51cto              51CTO           http://www.51cto.com/   
572  accelerator-corp  Accelerator Corp.  http://www.acceleratorcorp.com   

    category_list     status country_code state_code       region  \
30            NaN  operating          CHN          2     Hangzhou   
63            NaN  operating          CHN         22      Beijing   
207           NaN  operating          USA         CA  Los Angeles   
285           NaN  operating          CHN         22      Beijing   
572           NaN  operating          USA         WA      Seattle   

                city  founded_at company_permalink  \
30          Hangzhou  01-01-2014       123feng-com   
63           Beijing  01-01-2014           1mxians   
207  Manhattan Beach  01-01-2006       3pl-central   
285          Beijing         NaN             51cto   
572          Seattle  01-01-2003  accelerator-corp   

                             funding_round_permalink funding_round_type  \
30   /funding-round/9d83084d87dc63a309c9a6fe9bf59d1b            venture   
63   /funding-round/6950448f81c6706abe61f659e15a7c87            venture   
207  /funding-round/895e7a284b35bdf86103e52db8a630a6            venture   
285  /funding-round/8e4af9cbe06e0c5883b90d11a7c6251c            venture   
572  /funding-round/215f281607a54b9d5fc848b5c74ef9e7            venture   

    funding_round_code   funded_at  raised_amount_usd  
30                   A  27-10-2015         13000000.0  
63                   A  02-12-2015         15000000.0  
207                NaN  03-12-2015         17500000.0  
285                NaN  23-11-2015         16000000.0  
572                NaN  22-05-2003         15000000.0

In [22]:
#fill the NaNs with the text Blanks
top9['category_list'].fillna('NaN',inplace=True)

C:\Users\300236145\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\generic.py:6245: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._update_inplace(new_data)


In [23]:
len(top9[top9['category_list'].isna()])

0

In [24]:
#consider only the first sub category
top9['category_list'] = top9['category_list'].apply(lambda x: x.split('|')[0])

C:\Users\300236145\AppData\Local\Continuum\anaconda3\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [25]:
#now for Local Businesses|Restaurants we should get Local Businesses as the sub sector
top9['category_list'][:20]

0                Media
2          Curated Web
3        Biotechnology
4            Analytics
5            Analytics
6            Analytics
7                 Apps
8            Education
9          Curated Web
10         Curated Web
11         Curated Web
12         Curated Web
13         Curated Web
14    Local Businesses
15    Local Businesses
16    Local Businesses
17           Cosmetics
18               Games
19         Hospitality
20           Education
Name: category_list, dtype: object

In [26]:
len(top9['category_list'].unique())

577

***
<b> Now the number of unique sub sectors has reduced from earlier 8350 to only 577. These need to be mapped now to the major sectors as in the mapping file. </b>
***

In [27]:
top9['category_list'].value_counts().sort_values(ascending=False)

Biotechnology            6054
Software                 3554
Advertising              1937
Enterprise Software      1623
Health Care              1617
                         ... 
Mass Customization          1
Online Identity             1
Limousines                  1
Enterprise Purchasing       1
Tea                         1
Name: category_list, Length: 577, dtype: int64

In [28]:
#merge the top9 and mapping file and retain all the rows of top9
top9_merge = top9.merge(mapping,left_on='category_list',right_on='category_list',
                    how='left',suffixes=('_top9','_mapping'))

In [29]:
top9_merge.columns

Index(['permalink', 'name', 'homepage_url', 'category_list', 'status',
       'country_code', 'state_code', 'region', 'city', 'founded_at',
       'company_permalink', 'funding_round_permalink', 'funding_round_type',
       'funding_round_code', 'funded_at', 'raised_amount_usd',
       'Automotive & Sports', 'Blanks', 'Cleantech / Semiconductors',
       'Entertainment', 'Health', 'Manufacturing',
       'News, Search and Messaging', 'Others',
       'Social, Finance, Analytics, Advertising'],
      dtype='object')

In [30]:
#check the info
top9_merge.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 41526 entries, 0 to 41525
Data columns (total 25 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   permalink                                41526 non-null  object 
 1   name                                     41526 non-null  object 
 2   homepage_url                             39635 non-null  object 
 3   category_list                            41526 non-null  object 
 4   status                                   41526 non-null  object 
 5   country_code                             41526 non-null  object 
 6   state_code                               41257 non-null  object 
 7   region                                   41107 non-null  object 
 8   city                                     41107 non-null  object 
 9   founded_at                               34687 non-null  object 
 10  company_permalink                        41526

Now there are additional columns added which represent the major sectors. 

In [31]:
top9_merge.head()

permalink                    name          homepage_url  \
0                   fame                   #fame    http://livfame.com   
1                0-6-com                 0-6.com    http://www.0-6.com   
2  0ndine-biomedical-inc  Ondine Biomedical Inc.  http://ondinebio.com   
3                 0xdata                  H2O.ai        http://h2o.ai/   
4                 0xdata                  H2O.ai        http://h2o.ai/   

   category_list     status country_code state_code       region  \
0          Media  operating          IND         16       Mumbai   
1    Curated Web  operating          CHN         22      Beijing   
2  Biotechnology  operating          CAN         BC    Vancouver   
3      Analytics  operating          USA         CA  SF Bay Area   
4      Analytics  operating          USA         CA  SF Bay Area   

            city  founded_at      company_permalink  \
0         Mumbai         NaN                   fame   
1        Beijing  01-01-2007                0-6-com   
2      Vancouver  01-01-1997  0ndine-biomedical-inc   
3  Mountain View  01-01-2011                 0xdata   
4  Mountain View  01-01-2011                 0xdata   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
2  /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
3  /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
4  /funding-round/ae2a174c06517c2394aed45006322a7e            venture   

  funding_round_code   funded_at  raised_amount_usd  Automotive & Sports  \
0                  B  05-01-2015         10000000.0                  0.0   
1                  A  19-03-2008          2000000.0                  0.0   
2                NaN  21-12-2009           719491.0                  0.0   
3                  B  09-11-2015         20000000.0                  NaN   
4                NaN  03-01-2013          1700000.0                  NaN   

   Blanks  Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
0     0.0                         0.0            1.0     0.0            0.0   
1     0.0                         0.0            0.0     0.0            0.0   
2     0.0                         1.0            0.0     0.0            0.0   
3     NaN                         NaN            NaN     NaN            NaN   
4     NaN                         NaN            NaN     NaN            NaN   

   News, Search and Messaging  Others  Social, Finance, Analytics, Advertising  
0                         0.0     0.0                                      0.0  
1                         1.0     0.0                                      0.0  
2                         0.0     0.0                                      0.0  
3                         NaN     NaN                                      NaN  
4                         NaN     NaN                                      NaN

***
<span style='color:Red'> There are rows under all the columns representing the major sectors as null values. This cannot be the NaN string as the data type of all these columns is float64. It could be possible that in the mapping master file the corresponding sub-sectors do not exist !!
***
    

In [32]:
#check one of the sub sector which does not have a major sector assigned
mapping[mapping['category_list']=='Analytics']

Empty DataFrame
Columns: [category_list, Automotive & Sports, Blanks, Cleantech / Semiconductors, Entertainment, Health, Manufacturing, News, Search and Messaging, Others, Social, Finance, Analytics, Advertising]
Index: []

In [33]:
#total number of null values corresponding to the sub sectors which are not matched
len(top9_merge[top9_merge['Automotive & Sports'].isna()]['category_list'])

2934

In [34]:
#which are these sub sectors
top9_merge[top9_merge['Automotive & Sports'].isna()]['category_list'].value_counts()

Analytics                          1395
Finance                             602
NaN                                 177
Nanotechnology                      138
Financial Services                  107
Big Data Analytics                  100
Career Management                    60
Cloud Management                     47
Event Management                     28
Contact Management                   24
Finance Technology                   24
IT Management                        21
Document Management                  20
Identity Management                  19
Business Analytics                   16
Investment Management                14
Digital Signage                      12
Risk Management                      10
Energy Management                     9
Professional Services                 8
Project Management                    7
Innovation Management                 7
Navigation                            6
Lead Management                       5
Natural Language Processing           5


***
<span style='color:Red'> Analytics, Finance, Financial Services, Big Data Analytics, Investment Managament, Mobile Analytics, Personal Finance, Predictive Analytics, Professional Networking can be assigned to <b>Social, Finance, Analytics, Advertising
***

In [35]:
top9_merge[top9_merge['category_list']=='Analytics'].head(2) 

permalink    name    homepage_url category_list     status country_code  \
3    0xdata  H2O.ai  http://h2o.ai/     Analytics  operating          USA   
4    0xdata  H2O.ai  http://h2o.ai/     Analytics  operating          USA   

  state_code       region           city  founded_at company_permalink  \
3         CA  SF Bay Area  Mountain View  01-01-2011            0xdata   
4         CA  SF Bay Area  Mountain View  01-01-2011            0xdata   

                           funding_round_permalink funding_round_type  \
3  /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
4  /funding-round/ae2a174c06517c2394aed45006322a7e            venture   

  funding_round_code   funded_at  raised_amount_usd  Automotive & Sports  \
3                  B  09-11-2015         20000000.0                  NaN   
4                NaN  03-01-2013          1700000.0                  NaN   

   Blanks  Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
3     NaN                         NaN            NaN     NaN            NaN   
4     NaN                         NaN            NaN     NaN            NaN   

   News, Search and Messaging  Others  Social, Finance, Analytics, Advertising  
3                         NaN     NaN                                      NaN  
4                         NaN     NaN                                      NaN

In [36]:
#check all the sub category_list in mapping which starts with A's
#mapping[mapping['category_list'].str.contains('A')]['category_list']
mapping[mapping['Social, Finance, Analytics, Advertising']==1].head(25)

category_list  Automotive & Sports  Blanks  \
4                     Accounting                    0       0   
6                   Ad Targeting                    0       0   
9                    Advertising                    0       0   
10         Advertising Exchanges                    0       0   
11          Advertising Networks                    0       0   
12         Advertising Platforms                    0       0   
22                      A0lytics                    0       0   
23                       Android                    0       0   
26    Anything Capital Intensive                    0       0   
37       Artificial Intelligence                    0       0   
41                      Auctions                    0       0   
44             Augmented Reality                    0       0   
48                           B2B                    0       0   
49          B2B Express Delivery                    0       0   
54                       Banking                    0       0   
58                      Big Data                    0       0   
59             Big Data A0lytics                    0       0   
60                       Billing                    0       0   
70                  BPO Services                    0       0   
71               Brand Marketing                    0       0   
73   Bridging Online and Offline                    0       0   
79             Business A0lytics                    0       0   
80          Business Development                    0       0   
81  Business Information Systems                    0       0   
82         Business Intelligence                    0       0   

    Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
4                            0              0       0              0   
6                            0              0       0              0   
9                            0              0       0              0   
10                           0              0       0              0   
11                           0              0       0              0   
12                           0              0       0              0   
22                           0              0       0              0   
23                           0              0       0              0   
26                           0              0       0              0   
37                           0              0       0              0   
41                           0              0       0              0   
44                           0              0       0              0   
48                           0              0       0              0   
49                           0              0       0              0   
54                           0              0       0              0   
58                           0              0       0              0   
59                           0              0       0              0   
60                           0              0       0              0   
70                           0              0       0              0   
71                           0              0       0              0   
73                           0              0       0              0   
79                           0              0       0              0   
80                           0              0       0              0   
81                           0              0       0              0   
82                           0              0       0              0   

    News, Search and Messaging  Others  \
4                            0       0   
6                            0       0   
9                            0       0   
10                           0       0   
11                           0       0   
12                           0       0   
22                           0       0   
23                           0       0   
26                           0       0   
37                           0     

<b> <span style='color:Red'> Analytics is misspelled as A0lytics !!

In [37]:
#replace A0lytics with Analytics -- > will take care of 1395 NaN values 
mapping.replace(to_replace='A0lytics',value='Analytics',inplace=True)

In [38]:
#check whether rightly replaced
mapping['category_list'][20:24]

20    Alter0tive Medicine
21                 Alumni
22              Analytics
23                Android
Name: category_list, dtype: object

In [39]:
#Also Business Analytics
mapping.replace(to_replace='Business A0lytics',value='Business Analytics',inplace=True)

In [40]:
#check for more such instanes where mapping has A0lytics
mapping['category_list'].str.contains('A0lytics').value_counts()

False    684
True       3
Name: category_list, dtype: int64

In [41]:
mapping[mapping['category_list'].str.contains('A0lytics',na=False)]

category_list  Automotive & Sports  Blanks  \
59     Big Data A0lytics                    0       0   
448      Mobile A0lytics                    0       0   
534  Predictive A0lytics                    0       0   

     Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
59                            0              0       0              0   
448                           0              0       0              0   
534                           0              0       0              0   

     News, Search and Messaging  Others  \
59                            0       0   
448                           0       0   
534                           0       0   

     Social, Finance, Analytics, Advertising  
59                                         1  
448                                        1  
534                                        1

In [42]:
#do all the spelling correction for Analytics in the mapping file. 
mapping.replace(to_replace='Big Data A0lytics',value='Big Data Analytics',inplace=True)
mapping.replace(to_replace='Mobile A0lytics',value='Mobile Analytics',inplace=True)
mapping.replace(to_replace='Predictive A0lytics',value='Predictive Analytics',inplace=True)

In [43]:
#check all where Analytics has been corrected
mapping[mapping['category_list'].str.contains('Analytics',na=False)]

category_list  Automotive & Sports  Blanks  \
22              Analytics                    0       0   
59     Big Data Analytics                    0       0   
79     Business Analytics                    0       0   
448      Mobile Analytics                    0       0   
534  Predictive Analytics                    0       0   

     Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
22                            0              0       0              0   
59                            0              0       0              0   
79                            0              0       0              0   
448                           0              0       0              0   
534                           0              0       0              0   

     News, Search and Messaging  Others  \
22                            0       0   
59                            0       0   
79                            0       0   
448                           0       0   
534                           0       0   

     Social, Finance, Analytics, Advertising  
22                                         1  
59                                         1  
79                                         1  
448                                        1  
534                                        1

In [44]:
top9_merge = top9.merge(mapping,left_on='category_list',right_on='category_list',
                    how='left',suffixes=('_top9','_mapping'))

In [45]:
top9_merge.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 41526 entries, 0 to 41525
Data columns (total 25 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   permalink                                41526 non-null  object 
 1   name                                     41526 non-null  object 
 2   homepage_url                             39635 non-null  object 
 3   category_list                            41526 non-null  object 
 4   status                                   41526 non-null  object 
 5   country_code                             41526 non-null  object 
 6   state_code                               41257 non-null  object 
 7   region                                   41107 non-null  object 
 8   city                                     41107 non-null  object 
 9   founded_at                               34687 non-null  object 
 10  company_permalink                        41526

In [46]:
mapping[mapping['category_list'].str.contains('Fi',na=False)]

category_list  Automotive & Sports  Blanks  \
242       Engineering Firms                    0       0   
268  Field Support Services                    0       0   
269            File Sharing                    0       0   
270                    Film                    0       0   
271       Film Distribution                    0       0   
272         Film Production                    0       0   
273                  Fi0nce                    0       0   
274       Fi0nce Technology                    0       0   
275      Fi0ncial Exchanges                    0       0   
276       Fi0ncial Services                    0       0   
277                 FinTech                    0       0   
278                 Fitness                    0       0   
516          Perso0l Fi0nce                    0       0   
628          Staffing Firms                    0       0   

     Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
242                           0              0       0              0   
268                           0              0       0              0   
269                           0              0       0              0   
270                           0              1       0              0   
271                           0              1       0              0   
272                           0              1       0              0   
273                           0              0       0              0   
274                           0              0       0              0   
275                           0              0       0              0   
276                           0              0       0              0   
277                           0              0       0              0   
278                           0              0       1              0   
516                           0              0       0              0   
628                           0              0       0              0   

     News, Search and Messaging  Others  \
242                           0       1   
268                           0       1   
269                           0       1   
270                           0       0   
271                           0       0   
272                           0       0   
273                           0       0   
274                           0       0   
275                           0       0   
276                           0       0   
277                           0       0   
278                           0       0   
516                           0       0   
628                           0       1   

     Social, Finance, Analytics, Advertising  
242                                        0  
268                                        0  
269                                        0  
270                                        0  
271                                        0  
272                                        0  
273                                        1  
274                                        1  
275                                        1  
276                                        1  
277                                        1  
278                                        0  
516                                        1  
628                                        0

Another Gold Mine, all the Finance is mispelled as Fi)ncial

In [106]:
mapping.replace(to_replace='Fi0ncial Services',value='Financial Services',inplace=True)
mapping.replace(to_replace='Fi0nce',value='Finance',inplace=True)
mapping.replace(to_replace='Perso0l Fi0nce',value='Personal Finance',inplace=True)
mapping.replace(to_replace='Fi0ncial Exchanges',value='Financial Exchanges',inplace=True)
mapping.replace(to_replace='Fi0nce Technology',value='Finance Technology',inplace=True)


In [107]:
top9_merge = top9.merge(mapping,left_on='category_list',right_on='category_list',
                    how='left',suffixes=('_top9','_mapping'))

In [108]:
top9_merge.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 41526 entries, 0 to 41525
Data columns (total 25 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   permalink                                41526 non-null  object 
 1   name                                     41526 non-null  object 
 2   homepage_url                             39635 non-null  object 
 3   category_list                            41526 non-null  object 
 4   status                                   41526 non-null  object 
 5   country_code                             41526 non-null  object 
 6   state_code                               41257 non-null  object 
 7   region                                   41107 non-null  object 
 8   city                                     41107 non-null  object 
 9   founded_at                               34687 non-null  object 
 10  company_permalink                        41526

Now there are around 700 values which are still not mapped to any sector.

In [109]:
mapping[mapping['category_list'].str.contains('Pro',na=False)]

category_list  Automotive & Sports  Blanks  \
78              Building Products                    0       0   
83          Business Productivity                    0       0   
235    Employer Benefits Programs                    0       0   
272               Film Production                    0       0   
284               Food Processing                    0       0   
373         Intellectual Property                    0       0   
383    Internet Service Providers                    0       0   
407            Lifestyle Products                    0       0   
422              Loyalty Programs                    0       0   
437         Medical Professionals                    0       0   
472   Natural Language Processing                    0       0   
478       New Product Development                    0       0   
482                    Non Profit                    0       0   
541                   Procurement                    0       0   
542                Product Design                    0       0   
543  Product Development Services                    0       0   
544                  Productivity                    0       0   
545         Productivity Software                    0       0   
546        Professio0l Networking                    0       0   
547         Professional Services                    0       0   
548            Project Management                    0       0   
549                   Promotional                    0       0   
550           Property Management                    0       0   
590             Service Providers                    0       0   

     Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
78                            0              0       0              1   
83                            0              0       0              0   
235                           0              0       0              0   
272                           0              1       0              0   
284                           0              0       0              1   
373                           0              0       0              0   
383                           0              0       0              0   
407                           0              1       0              0   
422                           0              0       0              0   
437                           0              0       1              0   
472                           1              0       0              0   
478                           0              0       0              1   
482                           0              0       0              0   
541                           0              0       0              0   
542                           0              0       0              0   
543                           0              0       0              0   
544                           0              0       0              0   
545                           0              0       0              0   
546                           0              0       0              0   
547                           0              0       0              0   
548                           0              0       0              0   
549                           0              0       0              0   
550                           0              0       0              0   
590                           0              0       0              0   

     News, Search and Messaging  Others  \
78                            0       0   
83                            0       0   
235                           0       0   
272                           0       0   
284                           0       0   
373                           0       1   
383                           1       0   
407                           0       0   
422                           0       1   
437                           0       0   
472                           0       0   
478                        

In [110]:
mapping.replace(to_replace='Property Ma0gement',value='Property Management',inplace=True)
mapping.replace(to_replace='Promotio0l',value='Promotional',inplace=True)
mapping.replace(to_replace='Project Ma0gement',value='Project Management',inplace=True)
mapping.replace(to_replace='Professio0l Services',value='Professional Services',inplace=True)
mapping.replace(to_replace='0tural Language Processing',value='Natural Language Processing',inplace=True)
mapping.replace(to_replace='Medical Professio0ls',value='Medical Professionals',inplace=True)

In [111]:
top9_merge = top9.merge(mapping,left_on='category_list',right_on='category_list',
                    how='left',suffixes=('_top9','_mapping'))

In [112]:
top9_merge.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 41526 entries, 0 to 41525
Data columns (total 25 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   permalink                                41526 non-null  object 
 1   name                                     41526 non-null  object 
 2   homepage_url                             39635 non-null  object 
 3   category_list                            41526 non-null  object 
 4   status                                   41526 non-null  object 
 5   country_code                             41526 non-null  object 
 6   state_code                               41257 non-null  object 
 7   region                                   41107 non-null  object 
 8   city                                     41107 non-null  object 
 9   founded_at                               34687 non-null  object 
 10  company_permalink                        41526

In [113]:
mapping[mapping['category_list'].str.contains('Fle',na=False)]

category_list  Automotive & Sports  Blanks  \
281  Fleet Management                    0       0   

     Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
281                           0              0       0              0   

     News, Search and Messaging  Others  \
281                           0       1   

     Social, Finance, Analytics, Advertising  
281                                        0

In [114]:
mapping.replace(to_replace='Fleet Ma0gement',value='Fleet Management',inplace=True)

In [115]:
mapping[mapping['category_list'].str.contains('Ma0',na=False)]

category_list  Automotive & Sports  Blanks  \
91               Career Ma0gement                    0       0   
114               Cloud Ma0gement                    0       0   
145             Contact Ma0gement                    0       0   
198      Digital Rights Ma0gement                    0       0   
210            Document Ma0gement                    0       0   
240              Energy Ma0gement                    0       0   
256               Event Ma0gement                    0       0   
353            Identity Ma0gement                    0       0   
370          Innovation Ma0gement                    0       0   
372  Intellectual Asset Ma0gement                    0       0   
384          Investment Ma0gement                    0       0   
389                  IT Ma0gement                    0       0   
394           Knowledge Ma0gement                    0       0   
400                Lead Ma0gement                    0       0   
575                Risk Ma0gement                    0       0   
612        Social Media Ma0gement                    0       0   
633        Supply Chain Ma0gement                    0       0   
637                Task Ma0gement                    0       0   
672               Waste Ma0gement                    0       0   

     Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
91                            0              0       0              0   
114                           0              0       0              0   
145                           0              0       0              0   
198                           0              1       0              0   
210                           0              0       0              0   
240                           1              0       0              0   
256                           0              1       0              0   
353                           0              0       0              0   
370                           0              0       0              0   
372                           0              0       0              0   
384                           0              0       0              0   
389                           0              0       0              0   
394                           0              0       0              0   
400                           0              0       0              0   
575                           0              0       0              0   
612                           0              0       0              0   
633                           0              0       0              0   
637                           0              0       0              0   
672                           1              0       0              0   

     News, Search and Messaging  Others  \
91                            0       0   
114                           1       0   
145                           0       0   
198                           0       0   
210                           0       1   
240                           0       0   
256                           0       0   
353                           0       0   
370                           0       0   
372                           0       1   
384                           0       0   
389                           0       1   
394                           0       1   
400                           0       1   
575                           0       0   
612                           0       0   
633                           0       1   
637                           0       1   
672                           0       0   

     Social, Finance, Analytics, Advertising  
91                                         1  
114                                        0  
145                                        1  
198                                        0  
210                                        0  
240                                        0  
256                                        0  
3

In [116]:
#define a function to replace all Ma0gement with Management
def spell_correction(string):
    string = string.split()
    if string[-1] == 'Ma0gement':
        string[-1]= string[-1].replace('Ma0gement','Management')
    return ' '.join(string)
        

In [117]:
top9_merge[top9_merge['Automotive & Sports'].isna()]['category_list'].value_counts().sort_values(ascending=False)

NaN                                177
Nanotechnology                     138
Career Management                   60
Cloud Management                    47
Event Management                    28
Contact Management                  24
IT Management                       21
Document Management                 20
Identity Management                 19
Investment Management               14
Digital Signage                     12
Risk Management                     10
Energy Management                    9
Innovation Management                7
Navigation                           6
Lead Management                      5
Digital Rights Management            5
Supply Chain Management              5
Cannabis                             5
Intellectual Asset Management        4
Educational Games                    4
Knowledge Management                 4
Professional Networking              3
Journalism                           3
Personalization                      3
China Internet           

In [119]:
mapping[mapping['category_list'].str.contains('tech',na=False)]

category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
66   Biotechnology                    0       0                           1   
471  0notechnology                    0       0                           1   

     Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
66               0       0              0                           0       0   
471              0       0              0                           0       0   

     Social, Finance, Analytics, Advertising  
66                                         0  
471                                        0

In [120]:
mapping.replace(to_replace='0notechnology',value='Nanotechnology',inplace=True)

In [121]:
top9_merge = top9.merge(mapping,left_on='category_list',right_on='category_list',
                    how='left',suffixes=('_top9','_mapping'))

In [122]:
top9_merge.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 41526 entries, 0 to 41525
Data columns (total 25 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   permalink                                41526 non-null  object 
 1   name                                     41526 non-null  object 
 2   homepage_url                             39635 non-null  object 
 3   category_list                            41526 non-null  object 
 4   status                                   41526 non-null  object 
 5   country_code                             41526 non-null  object 
 6   state_code                               41257 non-null  object 
 7   region                                   41107 non-null  object 
 8   city                                     41107 non-null  object 
 9   founded_at                               34687 non-null  object 
 10  company_permalink                        41526

Now there are close to 500 rows which are still not mapped to one of the 8 major sectors. Most of these have spelling mistakes where 'Management' is spelled as 'Ma0gement'

In [123]:
top9_merge[top9_merge['Automotive & Sports'].isna()]['category_list'].value_counts().sort_values(ascending=False)

NaN                                177
Career Management                   60
Cloud Management                    47
Event Management                    28
Contact Management                  24
IT Management                       21
Document Management                 20
Identity Management                 19
Investment Management               14
Digital Signage                     12
Risk Management                     10
Energy Management                    9
Innovation Management                7
Navigation                           6
Digital Rights Management            5
Cannabis                             5
Supply Chain Management              5
Lead Management                      5
Educational Games                    4
Intellectual Asset Management        4
Knowledge Management                 4
Journalism                           3
Professional Networking              3
Personalization                      3
China Internet                       3
Waste Management         

In [182]:
mapping.replace(to_replace='Career Ma0gement',value='Career Management',inplace=True)
mapping.replace(to_replace='Cloud Ma0gement',value='Cloud Management',inplace=True)
mapping.replace(to_replace='Lead Ma0gement',value='Lead Management',inplace=True)
mapping.replace(to_replace='Contact Ma0gement',value='Contact Management',inplace=True)
mapping.replace(to_replace='IT Ma0gement',value='IT Management',inplace=True)
mapping.replace(to_replace='Task Ma0gement',value='Task Management',inplace=True)
mapping.replace(to_replace='Document Ma0gement',value='Document Management',inplace=True)
mapping.replace(to_replace='Identity Ma0gement',value='Identity Management',inplace=True)
mapping.replace(to_replace='Risk Ma0gement',value='Risk Management',inplace=True)
mapping.replace(to_replace='Energy Ma0gement',value='Energy Management',inplace=True)
mapping.replace(to_replace='Investment Ma0gement',value='Investment Management',inplace=True)
mapping.replace(to_replace='Energy Ma0gement',value='Energy Management',inplace=True)
mapping.replace(to_replace='Knowledge Ma0gement',value='Knowledge Management',inplace=True)
mapping.replace(to_replace='Intellectual Asset Ma0gement',value='Intellectual Asset Management',inplace=True)
mapping.replace(to_replace='Innovation Ma0gement',value='Innovation Management',inplace=True)
mapping.replace(to_replace='Event Ma0gement',value='Event Management',inplace=True)
mapping.replace(to_replace='Supply Chain Ma0gement',value='Supply Chain Management',inplace=True)
mapping.replace(to_replace='Digital Rights Ma0gement',value='Digital Rights Management',inplace=True)
mapping.replace(to_replace='Digital Sig0ge',value='Digital Signage',inplace=True)
mapping.replace(to_replace='Waste Ma0gement',value='Waste Management',inplace=True)
mapping.replace(to_replace='Professio0l Networking',value='Professional Networking',inplace=True)
mapping.replace(to_replace='Educatio0l Games',value='Educational Games',inplace=True)
mapping.replace(to_replace='Perso0l Branding',value='Personal Branding',inplace=True)
mapping.replace(to_replace='Perso0l Data',value='Personal Data',inplace=True)
mapping.replace(to_replace='Perso0lization',value='Personalization',inplace=True)
mapping.replace(to_replace='Perso0l Health',value='Personal Health',inplace=True)
mapping.replace(to_replace='Chi0 Internet',value='China Internet',inplace=True)
mapping.replace(to_replace='0tural Resources',value='Natural Resources',inplace=True)
mapping.replace(to_replace='Can0bis',value='Cannabis',inplace=True)
mapping.replace(to_replace='Jour0lism',value='Journalism',inplace=True)
mapping.replace(to_replace='Alter0tive Medicine',value='Alternative Medicine',inplace=True)

In [183]:
top9_merge = top9.merge(mapping,left_on='category_list',right_on='category_list',
                    how='left',suffixes=('_top9','_mapping'))

In [184]:
top9_merge.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 41526 entries, 0 to 41525
Data columns (total 25 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   permalink                                41526 non-null  object 
 1   name                                     41526 non-null  object 
 2   homepage_url                             39635 non-null  object 
 3   category_list                            41526 non-null  object 
 4   status                                   41526 non-null  object 
 5   country_code                             41526 non-null  object 
 6   state_code                               41257 non-null  object 
 7   region                                   41107 non-null  object 
 8   city                                     41107 non-null  object 
 9   founded_at                               34687 non-null  object 
 10  company_permalink                        41526

In [185]:
top9_merge[top9_merge['Automotive & Sports'].isna()]['category_list'].value_counts().sort_values(ascending=False)

NaN                                177
Navigation                           6
Adaptive Equipment                   2
Governance                           2
Natural Gas Uses                     1
Biotechnology and Semiconductor      1
Rapidly Expanding                    1
GreenTech                            1
Specialty Retail                     1
Name: category_list, dtype: int64

<b> <span style='color:Blue'> Barring 193 rows out of which 177 are mapped to NaN or major sectors as Blank, the remaining sub sectors can be ignored. </b>

In [202]:
null_index = top9_merge[top9_merge['Automotive & Sports'].isna()].index

In [203]:
len(null_index)

192

In [204]:
#drop these rows and we will also drop the Blanks column
top9_merge.drop(null_index,axis=0,inplace=True)

In [205]:
top9_merge.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 41334 entries, 0 to 41525
Data columns (total 25 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   permalink                                41334 non-null  object 
 1   name                                     41334 non-null  object 
 2   homepage_url                             39483 non-null  object 
 3   category_list                            41334 non-null  object 
 4   status                                   41334 non-null  object 
 5   country_code                             41334 non-null  object 
 6   state_code                               41066 non-null  object 
 7   region                                   40917 non-null  object 
 8   city                                     40917 non-null  object 
 9   founded_at                               34610 non-null  object 
 10  company_permalink                        41334

In [206]:
#drop the Blanks column as we only want to work with 8 major sectors
top9_merge.drop('Blanks',axis=1,inplace=True)

In [207]:
top9_merge.columns

Index(['permalink', 'name', 'homepage_url', 'category_list', 'status',
       'country_code', 'state_code', 'region', 'city', 'founded_at',
       'company_permalink', 'funding_round_permalink', 'funding_round_type',
       'funding_round_code', 'funded_at', 'raised_amount_usd',
       'Automotive & Sports', 'Cleantech / Semiconductors', 'Entertainment',
       'Health', 'Manufacturing', 'News, Search and Messaging', 'Others',
       'Social, Finance, Analytics, Advertising'],
      dtype='object')

In [208]:
top9_merge.to_csv('mainsectors.csv',index=False)